app/components/work_packages/share/modal_body_component.rb

```diff
@@ -30,14 +30,17 @@ module WorkPackages
   module Share
     class ModalBodyComponent < ApplicationComponent
       include ApplicationHelper
+      include MemberHelper
       include OpTurbo::Streamable
       include OpPrimer::ComponentHelpers
       include WorkPackages::Share::Concerns::Authorization
+      include WorkPackages::Share::Concerns::DisplayableRoles
 
-      def initialize(work_package:)
+      def initialize(work_package:, shares:)
         super
 
         @work_package = work_package
+        @shared_principals = shares
```

Comment:
I'd rename this to `@share` since now it no longer houses a `Principal` Relation but instead a `Member` relation

Reply:
mhh `@share` has led to some issues when inline creating as the same name is used there. I changed it now to `@shared_members` 🤷‍♀️